File: csharp_body_deref/Scripts/Gisketch.Aottg2UI.Game/AottgGuildPresetDefinition.cs
// ==================== AoTTG2 cross-reference ====================
// Type: Gisketch.Aottg2UI.Game.AottgGuildPresetDefinition
// Update status: unchanged since the previous dump
// Real method bodies (native C decompile): source/cpp/Gisketch/Aottg2UI/Game/AottgGuildPresetDefinition.c
// Prior real C# source: none (type is newer than the last public source snapshot)
// Bodies below are empty IL2CPP metadata stubs; the logic lives in the native C.
// ===============================================================
using System.Runtime.InteropServices;
using Il2CppDummyDll;

namespace Gisketch.Aottg2UI.Game;

[Token(Token = "0x20001A3")]
internal struct AottgGuildPresetDefinition
{
	[Token(Token = "0x4000928")]
	[Il2CppDummyDll.FieldOffset(Offset = "0x0")]
	public readonly string Id;

	[Token(Token = "0x4000929")]
	[Il2CppDummyDll.FieldOffset(Offset = "0x8")]
	public readonly string Label;

	[Token(Token = "0x400092A")]
	[Il2CppDummyDll.FieldOffset(Offset = "0x10")]
	public readonly AottgGuildPresetKind Kind;

	[Token(Token = "0x400092B")]
	[Il2CppDummyDll.FieldOffset(Offset = "0x18")]
	public readonly string Sprite;

	[Token(Token = "0x400092C")]
	[Il2CppDummyDll.FieldOffset(Offset = "0x20")]
	public readonly float ImageScale;

	[Token(Token = "0x60009E8")]
	[Address(RVA = "0x4419950", Offset = "0x4419950", VA = "0x4419950")]
	public AottgGuildPresetDefinition(string id, string label, AottgGuildPresetKind kind, [Optional] string sprite, float imageScale = 1f)
	{
	}
}

File: cpp/Gisketch/Aottg2UI/Game/AottgGuildPresetDefinition.c
// Type: Gisketch.Aottg2UI.Game.AottgGuildPresetDefinition
// Ghidra (System V/gcc) decompilation, IL2CPP structs + signatures applied.
// ---- AoTTG2 cross-reference ----
// Update status: unchanged since the previous dump
// C# structure: source/csharp/Scripts/Gisketch.Aottg2UI.Game/AottgGuildPresetDefinition.cs
// Prior real C# source: none
// --------------------------------

// Gisketch.Aottg2UI.Game.AottgGuildPresetDefinition$$.ctor
// il2cpp: void Gisketch_Aottg2UI_Game_AottgGuildPresetDefinition___ctor (Gisketch_Aottg2UI_Game_AottgGuildPresetDefinition_o __this, System_String_o* id, System_String_o* label, int32_t kind, System_String_o* sprite, float imageScale, const MethodInfo* method);
// 0x4519950

void Gisketch_Aottg2UI_Game_AottgGuildPresetDefinition___ctor
               (Gisketch_Aottg2UI_Game_AottgGuildPresetDefinition_o __this,System_String_o *id,
               System_String_o *label,int32_t kind,System_String_o *sprite,float imageScale,MethodInfo *method
               )

{
  undefined4 in_register_00000014;
  
  id->klass = (System_String_c *)label;
  il2cpp_runtime_helper_022b4080();
  id->monitor = (void *)CONCAT44(in_register_00000014,kind);
  il2cpp_runtime_helper_022b4080(&id->monitor,(void *)CONCAT44(in_register_00000014,kind));
  (id->fields)._stringLength = (int32_t)sprite;
  id[1].klass = (System_String_c *)method;
  il2cpp_runtime_helper_022b4080(id + 1,method);
  *(float *)&id[1].monitor = imageScale;
  return;
}


